account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py

```diff
@@ -29,11 +29,37 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff,
                 rec.debit_move_id.move_id if
                 rec.credit_move_id.journal_id.type == 'purchase' else
                 rec.credit_move_id)
+            lines = []
+            analytic_accounts = []
+
+            # We get the analytic account if this exists in the invoice
+            for line in invoice_move.move_id.line_ids.filtered(
+                    lambda r: r.analytic_account_id):
+                analytic_accounts.append(line.analytic_account_id.id)
+            total_analytic_account = len(analytic_accounts)
+            if total_analytic_account > 0:
+                analytic_accounts = list(set(analytic_accounts))[0]
+            else:
+                analytic_accounts = False
+            tax_accounts = []
+            taxes = self.env['account.tax'].search(
+                [('use_cash_basis', '=', True)])
+
+            # We get only the Profit / Loss line to put the analytic account
+            for tax in taxes:
+                tax_accounts.append(tax.cash_basis_account.id)
+                tax_accounts.append(tax.account_id.id)
+                tax_accounts.append(tax.refund_account_id.id)
+            for aml in move.line_ids:
+                if (aml.account_id.id not in tax_accounts and
+                        not aml.account_id.reconcile):
+                    lines.append((1, aml.id, {
+                        'analytic_account_id': analytic_accounts,
```

Comment:
try to send the empty list, if it fails you can use:

`analytic_accounts if analytic_accounts else False`